test/specs/lib/factories-test.tsx

```diff
@@ -138,6 +142,48 @@ const itOverridesDefaultPropsWithFalseyProps = (propsSource, shorthandConfig) =>
   })
 }
 
+const itHandlesTheElementBasedOnComponentType = () => {
```

Comment:
What is purpose of `itHandlesTheElementBasedOnComponentType`? As I see this code is not reusable, let's just use it in place.

Reply:
done, thanks